components/join-us-modal/join-us-modal.jsx

```diff
@@ -0,0 +1,332 @@
+    let formValues = this.state.formValues;
+
+    // if value of an image field is a link, we don't wanna include it in the formValues state
+    // as the link is just for previewing user's current profile and not the image object we are
+    // sending to backend
+    if (
+      field.type !== `image` ||
+      (field.type === `image` && typeof value !== `string`)
+    ) {
+      formValues[name] = value;
+      this.setState({ formValues });
+    }
+  }
+
+  handleFormSubmit() {
+    this.setState(
+      {
+        submitting: true
+      },
+      () => {
+        let formValues = this.state.formValues;
+        delete formValues[`email`]; // exclude email from data posting to API as email shouldn't be changeable
+
+        this.updateProfile(formValues);
+      }
+    );
+  }
+
+  updateProfile(formValues) {
+    Service.myProfile
+      .put(formValues)
+      .then(() => {
+        this.setState({ showConfirmation: true });
+      })
+      .catch(reason => {
+        this.setState({
+          // TODO:FIXME: do we care to have this state?
+          serverError: true
+        });
+        console.error(reason);
+      });
+  }
+
+  getNextScreenState() {
+    return this.state.currentStep < this.totalStep
+      ? { currentStep: this.state.currentStep + 1 }
+      : {};
+  }
+
+  handleNextBtnClick() {
+    // validate current form section
+    this.refs[`step${this.state.currentStep}Form`].validates(valid => {
+      console.log(`current form is valid? ${valid}`, this.state.formValues);
+
+      if (valid) {
+        if (this.state.currentStep === this.totalStep) {
+          // we've reached the last step of the form, submit the form
+          this.handleFormSubmit();
+        } else {
+          // more steps to complete
+          this.setState(this.getNextScreenState());
+        }
+      }
+    });
+  }
+
+  handleSkipBtnClick() {
+    let currentStep = this.state.currentStep;
+    let formValues = this.state.formValues;
+    let fieldsToReset = this.state.fields[`step${this.state.currentStep}`];
+
+    Object.keys(fieldsToReset).forEach(key => {
+      delete formValues[key];
+    });
+
+    this.setState(
+      Object.assign({ formValues }, this.getNextScreenState()),
+      () => {
+        if (currentStep === this.totalStep) {
+          this.handleFormSubmit();
+        }
+      }
+    );
+  }
+
+  renderProgressDots() {
+    let dots = [...Array(this.totalStep)].map((dot, i) => {
+      return (
+        <div
+          key={i}
+          className={`dot ${this.state.currentStep === i + 1 ? `filled` : ``}`}
+        />
+      );
+    });
+
+    return <div className="modal-progress">{dots}</div>;
+  }
+
+  renderSteps() {
+    return (
+      <div>
+        <Step
+          show={this.state.currentStep === 1}
+          heading="Complete your sign up"
+          subhead="Please confirm your information to finish signing up."
+          hint="(This information came from your Google or GitHub account)"
```

Comment:
can we know at this point whether this was github or google? Or do we only know "that this was a valid login"

Reply:
@gideonthomas ^ ?

Reply:
we'd have to change the api to support that feature. I think it's a lot of work for little value so I think that text currently suffices.